src/util/irep.h

```diff
@@ -104,6 +96,87 @@ const irept &get_nil_irep();
 /// * \ref irept::dt::sub : A vector of \ref irept which is used to store
 ///   ordered but unnamed children.
 ///
+/// * \ref ref_count : if sharing is activated, this is used to count the number
+///   of references to a node.
+///
+/// * \ref hash_code : if HASH_CODE is activated, this is used to cache the
+///   result of the hash function.
+template<typename treet>
+class tree_nodet
+{
+public:
+  // These are not stable.
+  typedef std::vector<treet> subt;
+
+  // named_subt has to provide stable references; with C++11 we could
+  // use std::forward_list or std::vector< unique_ptr<T> > to save
+  // memory and increase efficiency.
+
+#ifdef NAMED_SUB_IS_FORWARD_LIST
+  typedef std::forward_list<std::pair<irep_namet, irept>> named_subt;
```

Comment:
Please make sure you also test this with `NAMED_SUB_IS_FORWARD_LIST` being set. (Which obviously hasn't been done.)

Reply:
Yes, I guess that's a good occasion to add tests for that.